Pacemaker/clone_experiments/clone_resource.md

```diff
@@ -0,0 +1,346 @@
+# Objective
+  - Perform clone constrain related experiment like
+    - colocation
+    - location
+    - order
+  - Find multiple clone options and its meaning.
+  - Find multiple way to configure clone resource having multiple instances on node.
+  - Explore Group clone.
+
+# Summery
+  - By using clone constrain we can manipulate clone dependency and order among resources. Constrain and test cases given below.
+  - There and multiple clone option available from which we can manage clone size in cluster as well as per node, order of clone across nodes.
+  - We can configure clone resource having multiple instances as
+    - Create clone on cluster and divide in per node.
+    ```console
+    # Eg for 3 node and s3sever with 11 instance.
+    clone-max=33, clone-node-max=11
+    ```
+    - Create resource per node with max clone size per node.
+    ```console
+    # Eg for 3 node and s3sever with 11 instance.
+    For each resource on node clone-max=11, clone-node-max=11
+    resource will be fixed on that node.
+    ```
+    - Create clone group.
+    ```
+    # Eg for 3 node and s3sever with 11 instance.
+    Create 11 resource of s3server with group s3server_group and then clone that group.
+    ```
+  - We can use clone group property for io-stack and management stack.
+  - resource-discovery can be used to reduce network traffic in large cluster.
+
+# constraint
+  - location:
+    - prefers, avoid for normal case like allow/not allow resource on current node even if other node not available.
+    - complex: `constraint location csm-kibana rule score=-INFINITY '#uname' eq $lnode and consul-c1-running eq 0`
+    - Use **resource-discovery**=never if we don't want resource to run anywhere in system.
```

Comment:
"if we don't want resource to run anywhere in system" I am not sure if this statement is true. Please confirm.

Reply:
Modified description its already given below. It was like "Pacemaker will not discover resource on this node."

Reply:
ok